drivers/serial/uart_rpi_pico.c

```diff
@@ -402,8 +402,8 @@ static const struct uart_driver_api uart_rpi_driver_api = {
 	}									\
 										\
 	static const struct uart_rpi_config uart##idx##_rpi_config = {		\
-		.uart_dev = uart##idx,						\
-		.uart_regs = (uart_hw_t *)uart##idx,				\
+		.uart_dev = (uart_inst_t * const) DT_INST_REG_ADDR(idx),	\
```

Comment:
Please remove the space between the cast and the DT macro:
```suggestion
		.uart_dev = (uart_inst_t * const)DT_INST_REG_ADDR(idx),	\
```